src/freenet/clients/fcp/ClientGet.java

```diff
@@ -344,7 +355,7 @@ public void onLostConnection(ClientContext context) {
 	@Override
 	public void onSuccess(FetchResult result, ClientGetter state) {
 		Logger.minor(this, "Succeeded: "+identifier);
-		Bucket data = result.asBucket();
+		Bucket data = binaryBlob ? state.getBlobBucket() : result.asBucket();
```

Comment:
Given that it wasn't clear what this code does, it would be nice if you could add the answer to it as a comment. I've checked the replacement PR, and it doesn't seem to add a comment yet either: https://github.com/freenet/fred/pull/528/files